allennlp/data/samplers/samplers.py

```diff
@@ -0,0 +1,137 @@
+from typing import List, Iterable
+from torch.utils import data
+
+from allennlp.common.registrable import Registrable
+
+
+class Sampler(Registrable):
+    """
+    A copy of the pytorch [Sampler](https://pytorch.org/docs/stable/_modules/torch/utils/data/sampler.html)
+    which allows us to register it with `Registrable.`
+    """
+
+    def __iter__(self) -> Iterable[int]:
+
+        raise NotImplementedError
+
+
+class BatchSampler(Registrable):
+    """
+    A copy of the pytorch
+    [BatchSampler](https://pytorch.org/docs/stable/data.html#torch.utils.data.BatchSampler)
+    which allows us to register it with `Registrable.`
+    """
+
+    def __iter__(self) -> Iterable[List[int]]:
+
+        raise NotImplementedError
```

Comment:
Because they're abstract base classes, I guess? Should I change it to something else?

Reply:
Oh I see, you're asking why none of the subclasses implement them - it's because they all also inherit from the pytorch ones which do implement `__iter__` .